ingress/controllers/nginx/nginx/rewrite/main.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rewrite
+
+import (
+	"strconv"
+
+	"k8s.io/kubernetes/pkg/apis/extensions"
+)
+
+const (
```

Comment:
can you express this as a single annotation, they're tied together right? maybe just: `ingress.kubernetes.io/rewrite: {base, target}` ? you'd have to unmarshal a map but that's simple.

Also why `ingress-nginx`? just ignore the annotations you don't care about and use the rest? typically the first part of the annotations prefix is the resource name and has nothing to do with the implementation, and all site specific data is included after the "/"